tests/test_entities.py

```diff
@@ -2204,6 +2230,69 @@ def test_creation_and_update(self):
         self.assertEqual(expected_dct, cfg_template.update_payload())
 
 
+class TemplateInputTestCase(TestCase):
+    """Tests for :class:`nailgun.entities.TemplateInput`."""
+    def setUp(self):
+        self.entity = entities.TemplateInput(config.ServerConfig('some url'),
+                                             template='1')
+        self.read_patcher = mock.patch.object(entities.TemplateInput, 'read')
+        self.read = self.read_patcher.start()
+        self.read.return_value = entities.TemplateInput(
+            self.entity,
+            template=1,
+            name='my new template input',
+            input_type='user')
+        self.read_json_patcher = mock.patch.object(
+            EntityReadMixin,
+            'read_json')
+        self.read_json = self.read_json_patcher.start()
+
+    def tearDown(self):
+        self.read_json_patcher.stop()
+        self.read_patcher.stop()
+
+    def test_read(self):
+        entities.TemplateInput(
+            config.ServerConfig('some url'),
+            id='1',
+            name='my new template input',
+            template='1'
+        ).read(attrs={
+            'name': 'my new template input',
+            'template': '1',
+            'input_type': 'user'
+        })
+
+
+class JobTemplateTestCase(TestCase):
+    """Tests for :class:`nailgun.entities.JobTemplate`."""
+    def setUp(self):
+        self.entity = entities.JobTemplate(config.ServerConfig('some url'),
+                                           id='1')
+        self.read_patcher = mock.patch.object(EntityReadMixin, 'read')
+        self.read = self.read_patcher.start()
+        self.read.return_value = entities.JobTemplate(
+            self.entity,
+            id=1,
+            name='my new job template',
+            template='rm -rf /')
+        self.read_json_patcher = mock.patch.object(
+            EntityReadMixin, 'read_json')
+        self.read_json = self.read_json_patcher.start()
+
+    def tearDown(self):
+        self.read_json_patcher.stop()
+        self.read_patcher.stop()
+
+    def test_read(self):
+        entities.JobTemplate(config.ServerConfig('some url'), id='1').read(
+            attrs={'name': 'my new job template',
+                   'id': 1,
+                   'template': 'rm -rf /',
+                   'template_inputs': ''})
+        self.read.assert_called_once()
```

Comment:
we here are calling read so it will always succeed, I cannot get what exactly are testing here,
same reflection/ as previous test-case.

Reply:
suggest the following testcase for this entity read
```python

class JobTemplateTestCase(TestCase):
    """Tests for :class:`nailgun.entities.JobTemplate`."""
    def setUp(self):
        self.cfg = config.ServerConfig('some url')
        self.entity = entities.JobTemplate(self.cfg, id=1)
        self.read_json_patcher = mock.patch.object(EntityReadMixin, 'read_json')
        self.read_json = self.read_json_patcher.start()
        self.template_input_data = dict(id=1, template=1)
        self.data = dict(
            id=1,
            audit_comment=None,
            description_format=None,
            effective_user=None,
            job_category='Commands',
            location=[],
            locked=False,
            name='my new job template',
            organization=[],
            provider_type=None,
            snippet=False,
            template='rm -rf /',
            template_inputs=[self.template_input_data],
        )
        self.read_json.return_value = self.data.copy()
        del self.data['template_inputs']

    def tearDown(self):
        self.read_json_patcher.stop()

    def test_read(self):
        entity = self.entity.read()
        self.read_json.assert_called_once()
        self.assertEqual(
            self.data,
            {key: getattr(entity, key) for key in self.data.keys()}
        )
        self.assertEqual(len(entity.template_inputs), 1)
        template_input = entity.template_inputs[0]
        self.assertIsInstance(template_input, entities.TemplateInput)
        self.assertIsInstance(template_input.template, entities.JobTemplate)
        self.assertEqual(template_input.id,self.template_input_data['id'])
        self.assertEqual(template_input.template.id, self.template_input_data['template'])
```

Reply:
looks good, thank you!